deployment/puppet/cluster/files/q-agent-cleanup.py

```diff
@@ -477,6 +488,46 @@ def _reschedule_agent_l3(self, agent_type):
         self.log.info("_reschedule_agent_l3: ended rescheduling of orphaned routers")
         self.log.debug("_reschedule_agent_l3: end.")
 
+    def _reschedule_agent_lbaas(self, agent_type):
+        self.log.debug("_reschedule_agent_lbaas: start.")
+        alive_agents = []
+
+        # collect pool from removed lbaas agent
+        remove_pools = []
+        dead_agent = None
+        agent_host = socket.gethostname()
```

Comment:
是的  如果仿照l3 那样通过判断agent的live超时时间来迁移出这个执行stop机器上面的pool显然不行啊，
例如我在执行维护工作的时候，通过pcs ban掉某个节点， 但是在这个节点down掉后， 要等一段时间
才能看见这个lbaas agent是down掉的啊， 如果我们那个ocf stop执行的时候，刚好在那个等待时间内，
这时候不就是迁移不出去了吗  即使管理网络正常

2016-07-19 19:31 GMT+08:00 huntxu <notifications@github.com>:

> In deployment/puppet/cluster/files/q-agent-cleanup.py
> <https://github.com/eayunstack/fuel-library/pull/47#discussion_r71320885>:
>
> > @@ -477,6 +488,46 @@ def _reschedule_agent_l3(self, agent_type):
> >          self.log.info("_reschedule_agent_l3: ended rescheduling of orphaned routers")
> >          self.log.debug("_reschedule_agent_l3: end.")
> >
> > +    def _reschedule_agent_lbaas(self, agent_type):
> > +        self.log.debug("_reschedule_agent_lbaas: start.")
> > +        alive_agents = []
> > +
> > +        # collect pool from removed lbaas agent
> > +        remove_pools = []
> > +        dead_agent = None
> > +        agent_host = socket.gethostname()
>
> 迁移不出去？你是指在管理网络坏了的情况下？但还有很多管理网络没坏的情况下的维护工作啊，这个时候停止前迁移走pool就有意义了。
>
> 另外l3和dhcp的ocf，在start最后调用有个前提是if ! ocf_is_true
> "$OCF_RESKEY_multiple_agents"
> ，大概意思是在不允许多个agent同时工作的情况下，把所有支持的资源调度到一个agent上，这样子就事实上只有一个agent在工作了，不过貌似这个选项我们不会用，默认总是允许多个agent一起工作的，所以可以忽略
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eayunstack/fuel-library/pull/47/files/34227f5f94dbd51594b544910280d08652a6d3d8#r71320885>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGUT0a_Et5xqex4W2Dz9r3Q4AFyi8gXtks5qXLWugaJpZM4JOgK3>
> .
>